lab-kayla/lib/methods.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const List = module.exports = function() {
+  for(let key in arguments) {
+    this[key] = arguments[key];
+  }
+  this.length = arguments.length;
+};
+
+// O(n)
+List.prototype.copy = function() {
+  let result = new List();
+  for(let key in this) {
+    result[key] = this[key];
+  }
+  return result;
+};
+
+// O(n)
+List.prototype.push = function(value) {
+  let result = this.copy();
+  result[result.length++] = value;
+  return result;
+};
+
+// O(n)
+List.prototype.pop = function() {
+  let result = this.copy();
+  delete result[--result.length];
+
+  return {
+    value: this[this.length - 1],
+    list: result,
+  };
+};
+
+ //O(n)
+List.prototype.forEach = function(callback) {
+  for(let i = 0; i < this.length; i++) {
+    callback(this[i], i, this);
+  }
+};
+
+//O(n)
+List.prototype.filter = function(callback) {
+  let newVals = new List();
+  for(let i = 0; i < this.length; i++) {
+    let tf = callback(this[i], i, this);
+    if(tf === true) {
```

Comment:
Line 49 could just be written as ``if(tf)`` to check for the its presence as that's a truthy value.